packages/pipeline-editor/src/PipelineEditorWidget.tsx

```diff
@@ -1066,13 +1072,14 @@ export class PipelineEditor extends React.Component<
     // prepare pipeline submission details
     const pipelineFlow = this.canvasController.getPipelineFlow();
 
+    const runtime_config = dialogResult.value.runtime_config;
+    const runtime =
+      PipelineService.getRuntimeName(runtime_config, runtimes) || 'local';
```

Comment:
Does this mean that 'local' will be the default runtime so if no runtimes have been configured, they will ALWAYS be able to run 'local'?

Are there any places where the front-end looks up the runtime to get more information?

Reply:
This means that, if it can find a selection on the list of runtimes, uses `local` which is the only item that is inserted to the list on the UI. There is another call `PipelineService.getDisplayName`

Reply:
Right, but when there are no runtimes configured, 'local' will always be present - correct?

Does this work (to reduce hard-coding)?
```suggestion
      PipelineService.getRuntimeName(runtime_config, runtimes) || local_runtime.name;
```